pkg/sources/source.go

```diff
@@ -175,7 +175,7 @@ func (sp *SourceProcessor) Start(ctx context.Context) error {
 	var toVertexPartitionMap = make(map[string]int)
 	shuffleFuncMap := make(map[string]*shuffle.Shuffle)
 	for _, edge := range sp.VertexInstance.Vertex.Spec.ToEdges {
-		if edge.ToVertexType == dfv1.VertexTypeReduceUDF && edge.GetToVertexPartitionCount() > 1 {
+		if edge.GetToVertexPartitionCount() > 1 {
```

Comment:
Was this something we forgot to change when working on multi-partition support?

Reply:
we are adding message-id based shuffling for every vertex. this is to have deterministic writing when we have multi-partitioned edges and dedupe will work correctly because dedupe is per stream.